compiler/compile/OMRCompilation.cpp

```diff
@@ -815,14 +815,22 @@ OMR::Compilation::getOSRInductionOffset(TR::Node *node)
    if (osrNode->getOpCode().isCall())
       return 3;
 
-   switch (osrNode->getOpCodeValue())
+   // If the monent has a bytecode index of 0, it must be the
+   // monent for a synchronized method. A transition here
+   // must target the method entry.
+   if (osrNode->getOpCodeValue() == TR::monent)
       {
-      case TR::monent: return 1;
-      case TR::asynccheck: return 0;
-      default:
-         TR_ASSERT(0, "OSR points should only be calls, monents or asyncchecks");
+      if (osrNode->getByteCodeIndex() == 0)
```

Comment:
From my understanding of how monitor enters are currently implemented, there is no clear way to distinguish between the implicit and the explicit entries. A node flag exists, however, it is not reliable. Bytecode index is checked because an explicit monitor entry requires an argument and therefore couldn't be the first bytecode, only the implicit could be.